core/spi/src/main/java/org/wildfly/swarm/spi/api/MarkerContainer.java

```diff
@@ -32,12 +32,19 @@
 
     @SuppressWarnings("unchecked")
     default T addMarker(String markerName) {
-        addAsManifestResource(new StringAsset("marker"), ArchivePaths.create(PATH_MARKERS, markerName));
-
+        addMarker(this, markerName);
         return (T) this;
     }
 
     default boolean hasMarker(String markerName) {
-        return contains(ArchivePaths.create("META-INF/markers", markerName));
+        return hasMarker(this, markerName);
+    }
+
+    static void addMarker(ManifestContainer<?> manifestContainer, String markerName) {
```

Comment:
E.g. `WebArchive` mentioned in https://github.com/wildfly-swarm/wildfly-swarm/pull/459#discussion_r111376689.